src/firmware/linux/host/types.rs

```diff
@@ -250,3 +255,222 @@ impl SnpConfig {
         }
     }
 }
+
+/// Structure used for interacting with the Linux Kernel.
+///
+/// The original C structure looks like this:
+///
+/// ```C
+/// struct cert_table {
+///    struct {
+///       unsigned char guid[16];
+///       uint32 offset;
+///       uint32 length;
+///    } cert_table_entry[];
+/// };
+/// ```
+///
+#[derive(Clone, Default)]
+#[repr(C)]
+pub struct CertTableEntry {
+    /// Sixteen character GUID.
+    guid: [u8; 16],
+
+    /// The starting location of the certificate blob.
+    offset: u32,
+
+    /// The number of bytes to read from the offset.
+    length: u32,
+}
+
+#[repr(C)]
+pub struct CertTable {
+    /// Pointer to the Certificates in question.
+    entries: *const CertTableEntry,
+}
+
+impl Default for CertTable {
+    fn default() -> Self {
+        Self {
+            entries: &CertTableEntry::default() as *const CertTableEntry,
+        }
+    }
+}
+
+impl CertTable {
+    pub fn from_uapi(table: UapiCertTable) -> Self {
+        let mut entries: Vec<CertTableEntry> = vec![];
+        let mut tmp_guid: [u8; 16] = [0u8; 16];
+
+        for entry in table.entries {
+            // We have a problem if the GUID is anything other than 16 bytes.
+            assert_eq!(entry.guid().len(), 16);
```

Comment:
Given that `table` is a value provided as an argument, wouldn't it be better to change this function to return a `Result` and fail gracefully instead of using `assert`?

Reply:
I added in some error handling for this. Let me know what you think @slp.